source/common/stats/thread_local_store.h

```diff
@@ -503,8 +507,18 @@ class ThreadLocalStoreImpl : Logger::Loggable<Logger::Id::stats>, public StoreRo
 
   Thread::ThreadSynchronizer sync_;
   std::atomic<uint64_t> next_scope_id_{};
+  uint64_t next_histogram_id_ ABSL_GUARDED_BY(hist_mutex_) = 0;
 
   StatNameSetPtr well_known_tags_;
+
+  mutable Thread::MutexBasicLockable hist_mutex_;
+  StatSet<ParentHistogramImpl> histogram_set_ ABSL_GUARDED_BY(hist_mutex_);
+
+  // Histograms IDs that are queued for cross-scope release. Because there
+  // can be a large number of histograms, all of which are released at once,
+  // (e.g. when a scope is deleted), it is likely more efficient to batch their
+  // cleanup, which would otherwise entail a post() per histogram per thread.
+  std::vector<uint64_t> histograms_to_cleanup_ ABSL_GUARDED_BY(hist_mutex_);
```

Comment:
Do you think this is premature optimization? There aren't really that many histograms and scope deletion is not that common. I'm wondering if we can do away with this for now given how complicated this code already is? Wdyt?

Reply:
probably; I can back it out and save it to patch it in if there's any evidence of need.

I think a lot of the time scopes are not cleared en masse, but I remember at least one user had a pattern of using xDS where they basically burn down the entire world and bring it up again, though I don't know enough details of xDS tradeoffs.